src/api/app/controllers/person_controller.rb

```diff
@@ -134,6 +135,34 @@ def put_userinfo
     render_ok
   end
 
+  def get_watchlist
+    if @user == User.session
+      render xml: User.session.render_axml(render_watchlist_only: true)
+    else
+      render_error(status: 403, errorcode: 'anonymous_user',
+                   message: 'Anonymous user is not allowed here')
```

Comment:
If you are checking for nobody there is `.is_nobody?` or `.session!`. If not this error message is pretty misleading...